tensorflow/python/keras/mixed_precision/experimental/autocast_variable.py

```diff
@@ -185,46 +185,60 @@ def constraint(self):
     return self._variable.constraint
 
   def assign(self, value, use_locking=None, name=None, read_value=True):
-    return self._variable.assign(value, use_locking, name, read_value)
+    assign_op = self._variable.assign(value, use_locking, name, read_value)
+    return create_autocast_variable(assign_op) if read_value else assign_op
```

Comment:
I don't think it is great to call `create_autocast_variable()` here, but `self.__class__()`, `AutoCastVariable()` or `super(AutoCastVariable, self).__init__()` would not work for distributed variables since either inheritance is wrong or since `__init__()` would expect different arguments.

Reply:
I think subclasses just have to override this. We shouldn't enforce a contract on how to initialize subclasses.

Reply:
I agree. Thinking about this again, it is not too bad after all :+1: